content/docs/rfcs/36/README.md

```diff
@@ -0,0 +1,642 @@
+---
+slug: 35 title: 35/WAKU2-BINDINGS-API
+name: Waku v2 C Bindings API
+status: draft
+tags: waku-core
+editor: Richard Ramos <richard@status.im>
+contributors:
+- Franck Royer <franck@status.im>
+---
+
+# Introduction
+
+Native applications that wish to integrate Waku may not be able to use nim-waku and its JSON RPC API due to constraints
+on packaging, performance or executables.
+
+An alternative is to link existing Waku implementation as a static or dynamic library in their application.
+
+This specification describes the C API that SHOULD be implemented by native Waku library and that SHOULD be used to
+consume them.
+
+# Design requirements
+
+The API should be generic enough, so:
+
+- it can be implemented by both nim-waku and go-waku C-Bindings,
+- it can be consumed from a variety of languages such as C#, Kotlin, Swift, Rust, C++, etc.
+
+The selected format to pass data to and from the API is `JSON`.
+
+It has been selected due to its widespread usage and easiness of use. Other alternatives MAY replace it in the future (C
+structure, protobuf) if it brings limitations that need to be lifted.
+
+# The API
+
+## General
+
+### JsonResponse
```

Comment:
Shall we just call it `Response`?

Reply:
Changed my mind. let's keep the `Json` Prefix to specify `char *` in JSON format.